### DISCUSSION TITLE
Contributing non-trivial comments should give reputation

### DISCUSSION THREAD

**Discussion by DonDebonair** [dailydotdev/daily#1525] [Feature Requests]
Currently, it seems that the only way to get more reputation, is if comments you write are upvoted or when you submit stories. The problem with this is that reputation becomes a bit of a popularity contest. I think the goal should be to drive engagement, even if people disagree with what you're writing.

So my proposal: writing comments should give reputation. Of course, this needs some safeguards so it cannot be abused by just spamming comments, but the general idea is that if someone contributes meaningful comments, even if others are not upvoting it (because they don't agree, or because there are just many comments so yours might be overlooked), you can build up reputation.

**Comment by nimrodkra** [comment]
Thanks for the feedback, how would you suggest to evaluate if a comment is
meaningful or not?

At the moment we use social proof (reputation) for it. The main
advantage of this system is that it makes abusing and spamming a lot harder
and also guides behavior towards meaningful contributions (the
definition of meaningful is subjective). The disadavatges are clear as you
rightfully stated them.

We welcome any idea that would encourage meaningful contributions and will
prevent abusing and spam altogether.

**Reply by DonDebonair** [reply]
Using reputation to evaluate if a comment is meaningful creates a chicken/egg problem, doesn't it? You can only get reputation if people upvote your comments, and people will only see your comments if you have enough reputation. Do I understand that correctly?

My suggestion would be to give some baseline reputation for comments above a certain word count, but remove that reputation if those comments are downvoted below 0. This way, spam still gets punished, but people that contribute meaningful comments get reputation, even if their comments are not surfaced by your algorithm.

**Comment by nimrodkra** [comment]
Every new user on daily.dev starts with 10 reputation points, which is
enough for your comments to be visible right from the start. You don’t need
to accumulate more reputation for your comments to be seen by others.

There are also multiple ways to earn reputation beyond just commenting. I
recommend checking the docs to explore different ways to build your
reputation organically across the platform.

Longer comments aren’t necessarily more valuable, and promoting them just
to increase reputation opens the door for abuse. When building a platform,
it’s all about tradeoffs. Unfortunately, bad actors are a given, and any
system we put in place has to ensure it doesn’t make it easy for such users
to negatively impact the experience for everyone else.

Thanks for the feedback and for putting so much thought into it—it’s
genuinely valuable as we work to make daily.dev better for everyone.

**Reply by DonDebonair** [reply]
Thanks for your thoughtful responses! What you're saying makes a lot of sense. I will peruse the documentation to figure out how to get a higher reputation :)
